cmd/kubeadm/app/phases/upgrade/staticpods.go

```diff
@@ -192,3 +249,64 @@ func rollbackOldManifests(oldManifests map[string]string, origErr error, pathMgr
 	// Let the user know there we're problems, but we tried to reçover
 	return fmt.Errorf("couldn't upgrade control plane. kubeadm has tried to recover everything into the earlier state. Errors faced: %v", errs)
 }
+
+type etcdVersionResponse struct {
```

Comment:
this copies code from `app/preflight`?
Please move this code into `app/util/etcd` or even better, use the etcd Go client if possible / easy to accomplish

Reply:
it seems etcd go-client does not offer func to get version, at least I could not find it. If you do not mind I will just move it to app/util/etcd

Reply:
done